packages/ws-server/src/webSocketServer.ts

```diff
@@ -135,7 +145,13 @@ app.ws.use(async (ctx) => {
         return;
       }
 
-      // TODO: verify rate limit
+      // verify rate limit for batch_request method based on IP
+      if (limiter.shouldRateLimitOnMethod(ctx.ip, WS_CONSTANTS.BATCH_REQUEST_METHOD_NAME, ctx.websocket.requestId)) {
```

Comment:
Tests?
I think the relay has a test setup that you can mimic e.g. set the rate limit small and send requests to exceed it and verify